src/serialization/sb2.js

```diff
@@ -501,12 +516,14 @@ const parseBlock = function (sb2block, addBroadcastMsg, getVariableId, extension
         return;
     }
     const oldOpcode = sb2block[0];
+
     // If the block is from an extension, record it.
-    const dotIndex = blockMetadata.opcode.indexOf('.');
-    if (dotIndex >= 0) {
-        const extension = blockMetadata.opcode.substring(0, dotIndex);
-        extensions.extensionIDs.add(extension);
+    const index = blockMetadata.opcode.indexOf('_');
+    const prefix = blockMetadata.opcode.substring(0, index);
+    if (CORE_EXTENSIONS.indexOf(prefix) === -1) {
+        extensions.extensionIDs.add(prefix);
```

Comment:
This seems like the opposite of what this piece was previously doing -- am I misunderstanding something? We're only adding the extension if it's not one of the core?

Reply:
From conversation. This checks to see if the opcode prefix is *not* core. If it is not, then it attempts to load an extension using the opcode prefix.